lib/sync.js

```diff
@@ -545,13 +545,22 @@ SyncApi.prototype._sync = function(syncOptions) {
         console.error("/sync error %s", err);
         console.error(err);
 
-        debuglog("Starting keep-alive");
-        self._syncConnectionLost = true;
-        self._startKeepAlives().done(function() {
+        if (!self._syncConnectionLost) {
+            // This is the first failure, which may be spurious. To avoid unnecessary
```

Comment:
"which may be spurious" - Can you give examples on this please as per RL discussion. Namely that things like restarting servers / load balancers / etc will knife any ongoing connection, and we want to effectively ignore that case.